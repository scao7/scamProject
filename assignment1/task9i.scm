;assue iterative way is allow to use while loop
(define (ramanujani depth)
	(define i depth)
	;(define result 0.0)
	(define (iter result i) 
		;(>= i 0)	
		;(set! result )
		;(set! i (- i 1))
		(if (>= i 0)
			(iter (* (sqrt (+ result (+ 6 i) )) (+ i 1)) (- i 1))
			result
		)
		;result
	)
	(iter 0.0 depth)
)                  

(define(main) 
	(setPort (open (getElement ScamArgs 1) 'read))
	(define arg (readExpr))
	(println "(ramanujani " arg ") is " (fmt "%.25f" (ramanujani  arg)))
	
	;change format to pass test case
)
