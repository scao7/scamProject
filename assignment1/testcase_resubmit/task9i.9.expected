(ramanujani 12) is 3.9998695293549548779310499
$4$
